flow/layers/opacity_layer_unittests.cc

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/flow/layers/opacity_layer.h"
+
+#include "flutter/flow/testing/layer_test.h"
+#include "flutter/flow/testing/mock_layer.h"
+#include "flutter/fml/macros.h"
+#include "flutter/testing/mock_canvas.h"
+
+namespace flutter {
+namespace testing {
+
+using OpacityLayerDeathTest = LayerTest;
+using OpacityLayerTest = LayerTest;
+
+TEST_F(OpacityLayerDeathTest, LeafLayer) {
+  auto layer =
+      std::make_shared<OpacityLayer>(SK_AlphaOPAQUE, SkPoint::Make(0.0f, 0.0f));
+
+  EXPECT_DEATH_IF_SUPPORTED(layer->Preroll(preroll_context(), SkMatrix()),
+                            "layers\\(\\)\\.size\\(\\)\\ > 0");
+}
+
+TEST_F(OpacityLayerDeathTest, EmptyLayer) {
+  auto mock_layer = std::make_shared<MockLayer>(SkPath());
+  auto layer =
+      std::make_shared<OpacityLayer>(SK_AlphaOPAQUE, SkPoint::Make(0.0f, 0.0f));
+  layer->Add(mock_layer);
+
+  layer->Preroll(preroll_context(), SkMatrix());
+  EXPECT_EQ(mock_layer->paint_bounds(), SkPath().getBounds());
+  EXPECT_EQ(layer->paint_bounds(), mock_layer->paint_bounds());
+  EXPECT_FALSE(mock_layer->needs_painting());
+  EXPECT_FALSE(layer->needs_painting());
+
+  EXPECT_DEATH_IF_SUPPORTED(layer->Paint(paint_context()),
+                            "needs_painting\\(\\)");
+}
+
+TEST_F(OpacityLayerDeathTest, PaintBeforePreroll) {
+  SkPath child_path;
+  child_path.addRect(5.0f, 6.0f, 20.5f, 21.5f);
+  auto mock_layer = std::make_shared<MockLayer>(child_path);
+  auto layer =
+      std::make_shared<OpacityLayer>(SK_AlphaOPAQUE, SkPoint::Make(0.0f, 0.0f));
+  layer->Add(mock_layer);
+
+  EXPECT_DEATH_IF_SUPPORTED(layer->Paint(paint_context()),
+                            "needs_painting\\(\\)");
+}
+
+TEST_F(OpacityLayerTest, FullyOpaque) {
+  const SkPoint child_offset = SkPoint::Make(0.5f, 1.5f);
+  const SkRect child_bounds = SkRect::MakeLTRB(5.0f, 6.0f, 20.5f, 21.5f);
+  const SkPath child_path = SkPath().addRect(child_bounds);
+  const SkMatrix child_transform =
+      SkMatrix::MakeTrans(child_offset.fX, child_offset.fY);
+#ifndef SUPPORT_FRACTIONAL_TRANSLATION
```

Comment:
I don't think we have a variant that sets this. We should just remove it (I suppose in a later patch is fine).